kuttl-test.yaml

```diff
@@ -0,0 +1,5 @@
+apiVersion: kuttl.dev/v1beta1
```

Comment:
Some questions:
1. How it will support upgrade test where we need to start with previous operator version and then upgrade to image under test?
2. How it will support running the tests sometimes in cluster wide and sometimes in namespaced environment?
3. Do you see any problem like we had with diff files for openshift or different kubernetes distributions where we had to have different files?
4. Is it possible to define some global variables which will be available to all test steps? If I saw correctly you can access environment variables, but I'm not 100% sure.

Reply:
1. Either with kubernetes patch command or with new deployment with the same name but different image specified 
2. Only with additional efforts may be on bash side. TestsSuite support being executed on previously created namespace and dynamically created namespace inside the test. I believe we can add some magic here
3. At the moment I can't see issues since in our current bash framework platform specific diffs are the result of our yaml processor most of the time. Kuttl allows us to compare only really valued fields.
4. Yeap. ENV variables are accessible inside `script` directives.